controllers/datadogagent/common_rbac.go

```diff
@@ -161,3 +162,26 @@ func (r *Reconciler) cleanupServiceAccount(logger logr.Logger, client client.Cli
 	r.recordEvent(dda, event)
 	return reconcile.Result{}, client.Delete(context.TODO(), serviceAccount)
 }
+
+func (r *Reconciler) updateIfNeededClusterRoleBinding(logger logr.Logger, dda *datadoghqv1alpha1.DatadogAgent, clusterRoleBindingName, roleName, serviceAccountName, version string, clusterRoleBinding *rbacv1.ClusterRoleBinding) (reconcile.Result, error) {
+	info := roleBindingInfo{
+		name:               clusterRoleBindingName,
+		roleName:           roleName,
+		serviceAccountName: serviceAccountName,
+	}
+	newClusterRoleBinding := buildClusterRoleBinding(dda, info, version)
+	if !apiequality.Semantic.DeepEqual(newClusterRoleBinding.Subjects, clusterRoleBinding.Subjects) || !apiequality.Semantic.DeepEqual(newClusterRoleBinding.RoleRef, clusterRoleBinding.RoleRef) {
+		logger.V(1).Info("updateIfNeededClusterRoleBinding", "clusterRoleBinding.name", clusterRoleBinding.Name, "serviceAccount", serviceAccountName, "roleName", roleName)
+		// ClusterRoleBinding can be updated, if we change the RoleRef in it, we need to delete and recreate
+		if err := r.client.Delete(context.TODO(), clusterRoleBinding); err != nil {
+			return reconcile.Result{}, err
+		}
+		if err := r.client.Create(context.TODO(), newClusterRoleBinding); err != nil {
```

Comment:
Should we ignore the "not exists" error in the `Delete` above?

Not sure if the following could happen:
- Delete is successful
- Create fails because an error connection
- When the job is retried it will fail because the Delete will return "not found"

Reply:
If it was delete, we should not go again in the update function, but instead in the create function. So  I think it should be ok. Worth case it will take 2 reconcile
Loop

Reply:
:+1: